java/impl/gcp/src/main/java/co/worklytics/psoxy/SecretManagerConfigService.java

```diff
@@ -110,54 +108,39 @@ public Optional<String> getConfigPropertyAsOptional(ConfigProperty property) {
         }
     }
 
-    private String parameterName(ConfigProperty property) {
-        if (StringUtils.isBlank(this.namespace)) {
-            return property.name();
-        } else {
-            return this.namespace + property.name();
-        }
-    }
-
     @Override
     public boolean acquire(@NonNull String lockId, @NonNull java.time.Duration expires) {
         Preconditions.checkArgument(StringUtils.isNotBlank(lockId), "lockId cannot be blank");
 
-        final SecretName lockSecretName = SecretName.of(projectId, this.namespace + lockId);
-
-        long secretTtl = Math.max(expires.getSeconds(), LOCK_SECRETS_TTL.getSeconds());
+        final SecretName lockSecretName = getLockSecret(lockId);
 
         try (SecretManagerServiceClient client = SecretManagerServiceClient.create()) {
             Secret lockSecret;
             try {
                 lockSecret = client.getSecret(lockSecretName);
             } catch (com.google.api.gax.rpc.NotFoundException e) {
-                //create it
-                Secret initial = Secret.newBuilder()
-                    //ttl here, bc this Secret not managed by Terraform; so want it to clean itself
-                    // up after some time
-                    .setTtl(Duration.newBuilder().setSeconds(secretTtl).build())
-                    .build();
-                lockSecret = client.createSecret(lockSecretName.getProject(), lockSecretName.getSecret(),
-                    initial);
+                // It should not happen, as the secret has been created by Terraform
+                throw new RuntimeException(e);
```

Comment:
error message here should provide more detail on the problem; eg, secret named X must exist, should have been created by terraform.